tests/server/chunk_server/chunk_server_lease_impl_test.cc

```diff
@@ -0,0 +1,202 @@
+  GrantLeaseRequest req = MakeValidGrantLeaseRequest();
+  req.set_chunk_version(kTestFileVersion + 1);
+
+  grpc::ClientContext client_context;
+  auto reply_or = server_client_->SendRequest(req, client_context);
+  EXPECT_TRUE(reply_or.ok());
+  EXPECT_EQ(reply_or.ValueOrDie().status(),
+            GrantLeaseReply::REJECTED_STALE_VERSION);
+}
+
+TEST_F(ChunkServerLeaseImplTest, GrantLeaseStaleVersionInRequest) {
+  GrantLeaseRequest req = MakeValidGrantLeaseRequest();
+  req.set_chunk_version(kTestFileVersion - 1);
+
+  grpc::ClientContext client_context;
+  auto reply_or = server_client_->SendRequest(req, client_context);
+  EXPECT_TRUE(reply_or.ok());
+  EXPECT_EQ(reply_or.ValueOrDie().status(), GrantLeaseReply::UNKNOWN);
+}
+
+TEST_F(ChunkServerLeaseImplTest, GrantLeaseExpiredLease) {
+  GrantLeaseRequest req = MakeValidGrantLeaseRequest();
+  req.mutable_lease_expiration_time()->set_seconds(
+      absl::ToUnixSeconds(absl::Now() - absl::Minutes(1)));
+
+  grpc::ClientContext client_context;
+  auto reply_or = server_client_->SendRequest(req, client_context);
+  EXPECT_TRUE(reply_or.ok());
+  EXPECT_EQ(reply_or.ValueOrDie().status(),
+            GrantLeaseReply::IGNORED_EXPIRED_LEASE);
+}
+
+TEST_F(ChunkServerLeaseImplTest, RevokeLeaseValidRequest) {
+  RevokeLeaseRequest req = MakeValidRevokeLeaseRequest();
+  req.set_chunk_handle(kTestRevokeLeaseChunkHandle_REVOKABLE);
+
+  grpc::ClientContext client_context;
+  auto reply_or = server_client_->SendRequest(req, client_context);
+  EXPECT_TRUE(reply_or.ok());
+  EXPECT_EQ(reply_or.ValueOrDie().status(), RevokeLeaseReply::REVOKED);
+}
+
+TEST_F(ChunkServerLeaseImplTest, RevokeLeaseNewerLeaseAlreadyExists) {
+  RevokeLeaseRequest req = MakeValidRevokeLeaseRequest();
+  req.mutable_original_lease_expiration_time()->set_seconds(
+      kTestExpirationUnixSeconds - 1000);
+
+  grpc::ClientContext client_context;
+  auto reply_or = server_client_->SendRequest(req, client_context);
+  EXPECT_TRUE(reply_or.ok());
+  EXPECT_EQ(reply_or.ValueOrDie().status(),
+            RevokeLeaseReply::IGNORED_HAS_NEWER_LEASE);
+}
+
+TEST_F(ChunkServerLeaseImplTest, RevokeLeaseNoChunkHandle) {
+  RevokeLeaseRequest req = MakeValidRevokeLeaseRequest();
+  req.set_chunk_handle("no-such-handle");
+
+  grpc::ClientContext client_context;
+  auto reply_or = server_client_->SendRequest(req, client_context);
+  EXPECT_TRUE(reply_or.ok());
+  EXPECT_EQ(reply_or.ValueOrDie().status(),
+            RevokeLeaseReply::REJECTED_NOT_FOUND);
+}
+
+int main(int argc, char** argv) {
+  ::testing::InitGoogleTest(&argc, argv);
+
+  // Start a server for lease gRPC handler in the background, and wait some time
+  // for the server to successfully start in background
+  std::thread server_thread = std::thread(StartTestServer);
+  std::this_thread::sleep_for(std::chrono::seconds(3));
+
+  // Run tests
+  int exit_code = RUN_ALL_TESTS();
+
+  // Clean up background server
+  pthread_cancel(server_thread.native_handle());
```

Comment:
Not sure if this is the most elegant way to terminate a thread, but probably ok for a test.